lib/rsvp/utils.js

```diff
@@ -6,6 +6,10 @@ export function isFunction(x) {
   return typeof x === 'function';
 }
 
+export function isObject(x) {
+  return x !== undefined && x !== null && typeof x === 'object';
```

Comment:
my assumption was that checking for `undefined` is cheaper and `typeof` is expensive, so I am checking for undefined first (and if it is undefined it short circuits) am I wrong ?